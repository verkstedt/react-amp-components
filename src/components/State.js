/* eslint-disable react/no-danger */
import React, { Component, Fragment } from 'react'
import { shape, string } from 'prop-types'
import { Helmet } from 'react-helmet'

class State extends Component {
  static propTypes = {
    children: shape({}),
    id: string
  }

  static defaultProps = {
    children: {},
    id: null
  }

  render() {
    const { children, id } = this.props

    return (
      <Fragment>
        <Helmet key="helmet">
          <script async custom-element="amp-bind" src="https://cdn.ampproject.org/v0/amp-bind-0.1.js" />
        </Helmet>
        <amp-state id={id}>
          <script
            type="application/json"
            dangerouslySetInnerHTML={{
              __html: JSON.stringify(children)
            }}
          />
        </amp-state>
      </Fragment>
    )
  }
}

export default State
