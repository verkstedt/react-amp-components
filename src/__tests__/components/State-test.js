import React from 'react'
import State from '../../components/State'
import Helmet from '../../utils/Helmet'
import { renderComponent } from '../test-utils'

describe('State', () => {
  it('works', () => {
    const res = renderComponent(
      <State id="animals">
        {{
          lion: 'mammal'
        }}
      </State>
    )
    expect(res.toJSON()).toMatchSnapshot()
  })

  it('injects the right script tag', () => {
    renderComponent(
      <State>
        {{}}
      </State>
    )
    Helmet.canUseDOM = false
    const staticHead = Helmet.renderStatic()
    expect(staticHead.scriptTags).toMatchSnapshot()
  })
})

