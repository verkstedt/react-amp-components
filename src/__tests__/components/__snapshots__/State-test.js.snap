// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`State injects the right script tag 1`] = `
Array [
  Object {
    "async": true,
    "custom-element": "amp-bind",
    "src": "https://cdn.ampproject.org/v0/amp-bind-0.1.js",
  },
]
`;

exports[`State works 1`] = `
<amp-state
  id="animals"
>
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "{\\"lion\\":\\"mammal\\"}",
      }
    }
    type="application/json"
  />
</amp-state>
`;
